File: resto-pulse-mvp/backend/app/services/phone_tr.py
from __future__ import annotations

import re


def normalize_tr_mobile(raw: str | None) -> str | None:
    if not raw:
        return None
    digits = re.sub(r"\D", "", raw)
    if digits.startswith("90") and len(digits) >= 12:
        digits = digits[2:]
    if digits.startswith("0"):
        digits = digits[1:]
    if len(digits) == 10 and digits[0] == "5":
        return f"+90{digits}"
    return None


def is_tr_mobile(raw: str | None) -> bool:
    return normalize_tr_mobile(raw) is not None


def mask_phone_e164(phone_e164: str) -> str:
    digits = re.sub(r"\D", "", phone_e164)
    if len(digits) >= 4:
        return f"*** *** {digits[-2:]} {digits[-2:]}"
    return "***"

File: resto-pulse-mvp/mobile/lib/phone-tr.ts
/** Turk cep telefonu — backend normalize_tr_mobile ile uyumlu. */
export function normalizeTrMobileInput(raw: string): string | null {
  let digits = raw.replace(/\D/g, '');
  if (digits.startsWith('90') && digits.length >= 12) {
    digits = digits.slice(2);
  }
  if (digits.startsWith('0')) {
    digits = digits.slice(1);
  }
  if (digits.length === 10 && digits[0] === '5') {
    return `+90${digits}`;
  }
  return null;
}

export function formatTrMobileDisplay(e164: string): string {
  const digits = e164.replace(/\D/g, '');
  const local = digits.length >= 12 ? digits.slice(-10) : digits;
  if (local.length !== 10) return e164;
  return `0${local.slice(0, 3)} ${local.slice(3, 6)} ${local.slice(6, 8)} ${local.slice(8)}`;
}
